ClosedXML/Excel/Cells/XLCells.cs

```diff
@@ -148,11 +167,12 @@ private IEnumerable<XLSheetPoint> GetUsedCellsCandidates(XLWorksheet worksheet)
 
         public IEnumerator<XLCell> GetEnumerator()
         {
-            var cells = (_usedCellsOnly) ? GetUsedCells() : GetAllCells();
-            foreach (var cell in cells)
-            {
-                yield return cell;
-            }
+            return GetCells().GetEnumerator();
```

Comment:
reusing single implementation that doesn't require enumerator instances wrapping enumerators